src/main/kotlin/DirectoryLogic.kt

```diff
@@ -2,20 +2,57 @@ import java.lang.IllegalArgumentException
 
 class DirectoryLogic {
 
-    private val supportedExtensions = arrayOf("wav", "mp3", "jpeg", "tr")
+    private val supportedExtensions = arrayOf("wav", "mp3", "jpeg", "jpg", "tr")
+    private val supportedContainers = arrayOf("tr")
+    private val compressedTypes = arrayOf("mp3", "jpeg", "jpg")
+    private val groupings = arrayOf("book", "chapter", "verse", "chunk")
 
+    @Throws(IllegalArgumentException::class)
     fun buildFullFilePath(
-        inputFilePath: String,
-        languageCode: String,
-        dublinCoreId: String,
-        group: String,
-        projectId: String = "",
-        mediaExtension: String = "",
-        mediaQuality: String = "hi"
+            inputFilePath: String,
+            languageCode: String,
+            dublinCoreId: String,
+            group: String,
+            projectId: String = "",
+            mediaExtension: String = "",
+            mediaQuality: String = ""
     ): String {
 
-        return ""
+        val fileExt = inputFilePath.split("/").last().split('.').last()
+
+        var path = ""
+
+        if (languageCode.isBlank()) throw IllegalArgumentException("Language code is empty")
```

Comment:
Perhaps it would be better to validate the data first in a method call rather than scattered throughout the function.

That allows you to more easily test your data validation too.